mountpoint-s3/src/prefetch/backpressure_controller.rs

```diff
@@ -0,0 +1,118 @@
+use async_channel::{unbounded, Receiver, Sender};
+use tracing::trace;
+
+use super::PrefetchReadError;
+
+#[derive(Debug)]
+pub struct BackpressureController {
+    read_window_updater: Sender<usize>,
+    preferred_read_window_size: usize,
+    next_read_window_offset: u64,
+    last_request_offset: u64,
+}
+
+#[derive(Debug)]
+pub struct BackpressureLimiter {
+    read_window_incrementing_queue: Receiver<usize>,
+}
+
+/// Creates a [BackpressureController] and its related [BackpressureLimiter].
```

Comment:
Could we explain what's the function of these 2 types? Their methods and fields could also do with some comments (and/or clearer names).